### ISSUE TITLE
Consider including CKEditor as a git submodule

### ISSUE THREAD

**Issue by dandv** [liferay/alloy-editor#230]
Right now the CKEditor files are copied into the Alloy project. This makes it difficult for someone new to both to distinguish what's CK from what' Alloy.

Including the CKeditor as a git submodule would provide for a clear separation, and it would also make it easier to keep Alloy up to date with the upstream.

**Comment by ipeychev** [comment]
Hello,

We already considered that in the past and we even added it as submodule temporally. However, we later removed it and went with a custom build of CKEditor stored in `lib` for the following reasons:
- If we put the sources of CKEditor in the project, then we will have to build it. And we want to create a custom version, not to use their default configuration. However, CKEditor's build system is written in Java. Because of that, now the developers should have installed Java on their systems or we have to download it explicitly on the fly. This one we didn't want to do, we wanted to depend only on Node.
- Building of CKEditor takes ~30 sec on my machine (which is pretty fast). This one I didn't want to do every time we build AlloyEditor. In comparison, the build of AlloyEditor on the same machine takes about 2 sec. In watch mode - about 1.5sec.

Since we don't upgrade very often to another version of CKEditor (neither they release new versions often), we just created a custom build and we stored it it lib. Now everybody is happy - the build is fast and CKEditor is separated in `lib`.

Thanks,
